web/modules/custom/product_uploader/product_uploader.module

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @file
+ * This file adds/removes “Product uploader” role to the user upon registration.
+ *
+ * ...when the user’s email address contains/doesn't contain a “
+ * +productuploader” fragment (e.g. johndoe+productuploaded@gmail.com).
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ */
+function product_uploader_user_presave(EntityInterface $entity): void {
+  $fragment = '+productuploader';
```

Comment:
Move this to a constant outside of this function using the `const` keyword. Prefix the constant's name with the module's name. It should be something like `PRODUCT_UPLOADER_FRAGMENT`.